libdevcore/Visitor.h

```diff
@@ -20,109 +20,41 @@
 
 #pragma once
 
-#include <functional>
-#include <boost/variant/static_visitor.hpp>
-
 namespace dev
 {
 
-/// Generic visitor used as follows:
-/// std::visit(GenericVisitor<Class1, Class2>(
-///     [](Class1& _c) { _c.f(); },
-///     [](Class2& _c) { _c.g(); }
-/// ), variant);
-/// This one does not have a fallback and will fail at
-/// compile-time if you do not specify all variants.
-
-template <class...>
-struct GenericVisitor{};
-
-template <class Visitable, class... Others>
-struct GenericVisitor<Visitable, Others...>: public GenericVisitor<Others...>
-{
-	using GenericVisitor<Others...>::operator ();
-	explicit GenericVisitor(
-		std::function<void(Visitable&)> _visitor,
-		std::function<void(Others&)>... _otherVisitors
-	):
-		GenericVisitor<Others...>(std::move(_otherVisitors)...),
-		m_visitor(std::move(_visitor))
-	{}
-
-	void operator()(Visitable& _v) const { m_visitor(_v); }
-
-	std::function<void(Visitable&)> m_visitor;
-};
-template <>
-struct GenericVisitor<>: public boost::static_visitor<> {
-	void operator()() const {}
-};
-
-/// Generic visitor with fallback:
-/// std::visit(GenericFallbackVisitor<Class1, Class2>(
-///     [](Class1& _c) { _c.f(); },
-///     [](Class2& _c) { _c.g(); }
-/// ), variant);
-/// This one DOES have a fallback and will NOT fail at
-/// compile-time if you do not specify all variants.
-
-template <class...>
-struct GenericFallbackVisitor{};
-
-template <class Visitable, class... Others>
-struct GenericFallbackVisitor<Visitable, Others...>: public GenericFallbackVisitor<Others...>
-{
-	explicit GenericFallbackVisitor(
-		std::function<void(Visitable&)> _visitor,
-		std::function<void(Others&)>... _otherVisitors
-	):
-		GenericFallbackVisitor<Others...>(std::move(_otherVisitors)...),
-		m_visitor(std::move(_visitor))
-	{}
-
-	using GenericFallbackVisitor<Others...>::operator ();
-	void operator()(Visitable& _v) const { m_visitor(_v); }
-
-	std::function<void(Visitable&)> m_visitor;
-};
-template <>
-struct GenericFallbackVisitor<>: public boost::static_visitor<> {
-	template <class T>
-	void operator()(T&) const { }
-};
-
-/// Generic visitor with fallback that can return a value:
-/// std::visit(GenericFallbackReturnsVisitor<ReturnType, Class1, Class2>(
-///     [](Class1& _c) { return _c.f(); },
-///     [](Class2& _c) { return _c.g(); }
-/// ), variant);
-/// This one DOES have a fallback and will NOT fail at
-/// compile-time if you do not specify all variants.
-/// The fallback {}-constructs the return value.
-
-template <class R, class...>
-struct GenericFallbackReturnsVisitor{};
+/**
+ * Generic visitor used as follows:
+ * std::visit(GenericVisitor{
+ *     [](Class1& _c) { _c.f(); },
+ *     [](Class2& _c) { _c.g(); }
+ * }, variant);
+ * This one does not have a fallback and will fail at
+ * compile-time if you do not specify all variants.
+ *
+ * Fallback with no return:
```

Comment:
```suggestion
 * Fallback with no return (it will not fail if you do not specify all variants):
```
(basically explain what fallback means)